store/store.go

```diff
@@ -378,3 +378,28 @@ func (s *store) WriteBatch() error {
 func (s *store) IsBanned(addr crypto.Address) bool {
 	return s.config.BannedAddrs[addr]
 }
+
+func (s *store) pruneBlock(batch *leveldb.Batch, blockHeight uint32) (bool, error) { //nolint
+	s.lk.Lock()
+	defer s.lk.Unlock()
+
+	blkData, err := s.blockStore.block(blockHeight)
+	if err != nil {
+		return false, err
+	}
+
+	blk, err := block.FromBytes(blkData)
+	if err != nil {
+		return false, err
+	}
+
+	if err := s.blockStore.db.Delete(blk.Hash().Bytes(), nil); err != nil {
+		return false, err
+	}
+
+	for _, t := range blk.Transactions() {
+		batch.Delete(t.ID().Bytes())
+	}
+
```

Comment:
Please add `s.batch.Reset()` after prune block with transactions.

Reply:
Why batch should `Reset`?

Reply:
> Why batch should `Reset`?

I don't have good reason for this but check leveldb tests, after batch operation, reset used.

https://github.com/syndtr/goleveldb/blob/126854af5e6d8295ef8e8bee3040dd8380ae72e8/leveldb/batch_test.go#L87

Reply:
@Ja7ad Maybe they are using this to prevent conflicts in test(?).
We had other batch operations on the store without calling reset.

Reply:
We should have a reason for that.